docs/DeveloperGuide.md

```diff
@@ -1,38 +1,239 @@
+
+## Design & implementation
+
+### Architecture Component
+
+<div align="center">
+<img src="UML/Images/architecture.png"/>
+<p>
+Figure 1.1
+</p>
+</div>
+
+The architecture diagram as shown in Figure 1.1 shows us the high level design of Fitness Duke.
+
+**Main Components**
+
+The architecture comprises five main components.
+* ```main```: Responsible for the initialisation of user data, app data and connection of various components together
+* ```Ui```: The User Interface of the app
+* ```CommandHandler```: The main logic controller of the app
+* ```Storage```: Reading and Writing data to and from the hard disk
+
+```Hard Drive``` represents the files in which user data are stored
+
+**Interaction of components**
+
+<div align="center">
+<img src="UML/Images/overall_sequence-0.png"/>
+<p>
+Figure 1.2
+</p>
+</div>
+
+The sequence diagram as shown in Figure 1.2 shows the interaction of all the main components when a user enters the 
+```start``` command and subsequently enters the ```finish``` command to simulate the process of a workout session 
+experienced by the user.
+
+The five main components in the diagram shows the main classes participating in the process.
+
+These classes will be defined into its own API as an ```interface``` in the upcoming iteration
+
+More details on these components will be described below.
+
+### UI Component
+
+Represented by the Ui class, it handles the interactions between the user and the program. In addition, it also displays
+a basic overview of what the program does.
+
+Types of pre-defined messages:
+
+1. Information messages that describe the functionalities of the program (`Greet` , `Bye`, `PrintExercises` classes)
+2. Help messages that details the usage of the possible commands available in the program (`PrintHelpMessage` class)
+3. Error messages that trigger when a user's input is incorrect and provides an explanation to the
+   user. (`ErrorMessages` class)
+
+The class diagram as shown in Figure *1.3* illustrates the structure of the different classes in Ui.
+
+<div align="center">
+<img src="UML/Images/Ui.png"/>
+<p>
+Figure 1.3
+</p>
+</div>
+
+### Storage Component
+
+API: [```StorageHandler.java```](https://github.com/AY2223S2-CS2113-W13-2/tp/blob/master/src/main/java/seedu/duke/storage/StorageHandler.java)
+
+The *Storage* component handles the reading and writing of user data to and from the local hard disk in the form of
+a json file.
+
+Key Aspects:
+
+* Handles the serialization and deserialization of user data into a json file using
+  the [Gson Library](https://github.com/google/gson)
+* Handles the creation of user data file when previous one is missing or corrupted
+* Handles the loading of user data and plans upon start of the program
+
+The class diagram as shown in *Figure 1.4* illustrates the structure of the different classes in Storage.
+<div align="center">
+<img src="UML/Images/Storage.png"/>
+<p>
+Figure 1.4
+</p>
+</div>

```

Comment:
The `StorageManager` has an association with `UserPlansStorage` and `UserCareerStorage`, however this is might be already implied and not needed since `StorageManager` implements `Storage` which inherits from `UserPlansStorage` and `UserCareerStorage`.

![image](https://user-images.githubusercontent.com/74498975/228731129-439d027c-03a1-4600-a3ba-b1181ab4a0b0.png)